src/test.py

```diff
@@ -62,10 +65,105 @@ def wrap_socket(self, s, server_hostname):
     def patch(cls):
         return mock.patch("ssl.create_default_context", wraps=cls)
 
+class SilentTk:
+    def bind(self, event, callback):
+        pass
+
+tkinter.Tk = SilentTk
+
+class SilentCanvas:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def create_text(self, x, y, text, font=None, anchor=None):
+        pass
+
+    def pack(self):
+        pass
+
+    def delete(self, v):
+        pass
+
+tkinter.Canvas = SilentCanvas
+
+class MockCanvas:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def create_text(self, x, y, text, font=None, anchor=None):
+        if font or anchor:
+            print("create_text: x={} y={} text={} font={} anchor={}".format(
+                x, y, text, font, anchor))
+        else:
+            print("create_text: x={} y={} text={}".format(
+                x, y, text))
+
+    def pack(self):
+        pass
+
+    def delete(self, v):
+        pass
+
+original_tkinter_canvas = tkinter.Canvas
+
+def patch_canvas():
+    tkinter.Canvas = MockCanvas
+
+def unpatch_canvas():
+    tkinter.Canvas = original_tkinter_canvas
+
+class MockFont:
+    def __init__(self, size=None, weight=None, slant=None, style=None):
+        self.size = size
+        self.weight = weight
+        self.slant = slant
+        self.style = style
+
+    def measure(self, word):
+        return self.size * len(word)
+
+    def metrics(self, name=None):
+        all = {"ascent" : self.size / 2, "descent": self.size / 2,
+            "linespace": self.size * 2}
```

Comment:
This is all fine but I think in Tk, `linespace = ascent + descent`, it doesn't include suggested interline spacing, so this is a pretty weird font by Tk standards. Also `ascent = descent` is a bit unrealistic; it's usually something like `ascent = size * .75` and `descent = size * .25`. Again, this might not matter much for our tests, since they don't have to be super realistic, but it also doesn't seem like it could hurt?

Reply:
Done.